presto-native-execution/presto_cpp/main/types/PrestoToVeloxQueryPlan.cpp

```diff
@@ -112,6 +100,46 @@ std::shared_ptr<connector::ColumnHandle> toColumnHandle(
   throw std::invalid_argument("Unknown column handle type: " + column->_type);
 }
 
+connector::hive::LocationHandle::TableType toTableType(
+    protocol::TableType tableType) {
+  switch (tableType) {
+    case protocol::TableType::NEW:
+      return connector::hive::LocationHandle::TableType::kNew;
+    case protocol::TableType::EXISTING:
+      return connector::hive::LocationHandle::TableType::kExisting;
+    case protocol::TableType::TEMPORARY:
+      return connector::hive::LocationHandle::TableType::kTemporary;
+    default:
+      throw std::invalid_argument("Unknown table type");
```

Comment:
Might be better to throw velox exception from here.

Reply:
Got feedback on this in https://github.com/prestodb/presto/pull/18377, and address all occurrences in this file there.

Corresponding code copy of this PR has already been merged into fbcode repo. This PR is supposed to be just porting to GitHub, so better to keep the code identical of the fbcode copy. It's just Merge was blocked by the protobuf path issue as well as velox_hive_connector OBJECT issue as in https://github.com/facebookincubator/velox/pull/3094.

Reply:
@gggrace14 and everyone.
We do not want to port fbcode changes to GH. We want to avoid any changes in fbcode.
We will get this through, though.

@gggrace14 can you please create and follow up a small issue on changing this exception to Velox exception, please? To be consistent.